src/api/v2/resolvers/dpid.ts

```diff
@@ -148,8 +148,11 @@ export const resolveDpid = async (dpid: number, versionIx?: number): Promise<His
                 setToCache(streamCacheKey, "", CACHE_TTL_PENDING);
             }
         } else {
-            // The BigNumbers are parsed into objects, which ethers.BigNumber in fine with
             resolvedEntry = JSON.parse(fromCache);
+            // The BigNumbers are deserialized into objects, which ethers.BigNumber can instantiate from
+            resolvedEntry[1].forEach((v) => {
+                v[1] = BigNumber.from(v[1]);
+            });
```

Comment:
This is the meat: when deserialising from cache, instantiate as an actual `BigNumber` object